vscode/src/jsonrpc/agent-protocol.ts

```diff
@@ -44,6 +44,8 @@ export type Requests = {
     'graphql/getRepoIdIfEmbeddingExists': [{ repoName: string }, string | null]
     'graphql/getRepoId': [{ repoName: string }, string | null]
 
+    'git/codebaseName': [[cloneURL: string], string | null]
```

Comment:
Is there a reason this is an array instead of object?
```suggestion
    'git/codebaseName': [{cloneURL: string}, string | null]
```

Reply:
I was initially attempting to use `{ cloneURL: string }` as the argument type here. However, the issue was that despite the specified argument type (`string`), the method `convertGitCloneURLToCodebaseName(cloneURL: string)` was receiving an object instead. It caused `TypeError: undefined is not a function` at `cloneURL.replace(...)` call.

Reply:
I guess I can use an object, and change the argument type in jetbrains to smth like:
```java
public class CloneUrl {
  public final String url;

  public CloneUrl(String url) {
    this.url = url;
  }
}
```

Reply:
I did it like this for consistency.